extra/models/inception.py

```diff
@@ -301,3 +306,37 @@ def __call__(self, x:Tensor) -> Tensor:
       lambda x: Tensor.avg_pool2d(x, kernel_size=(8,8)),
     ])
     return x
+
+  def compute_score(self, inception_activations:Tensor, val_stats_path:str) -> float:
+    if self.m1 is None or self.s1 is None:
```

Comment:
it overwrites both m1, s1 even if one is None and another is not, this might cause silent issue. checking both are None to enter this path and check both are not None after this block

Reply:
I do not understand what your suggestion is. These are designed to be loaded in pairs, meaning if you load 1 you should be loading the other associated with it. Setting just 1 is invalid when you do not know where the other is coming from.

Reply:
`meaning if you load 1 you should be loading the other associated with it`. the implementation does not enforce this

Reply:
I am not following, the logic as it's written is that if either is None then set both of them and make sure they are both non-None afterwards. The code always loads both at the same time and makes sure we do not get into a weird state were halves are not properly loaded.

Reply:
If m1 and s1 will always both be None or both be non-None, then `if self.m1 is None and self.s1 is None:` is equivalent. It has the benefit to express what's the state expected here. We expect both to be None, not any of those to be None. So it reads weird to me that the starting condition of this block is different from the end condition

Also having the assert check outside of if block separate the setup from the validation.

Reply:
ok this makes sense, updated the code as this is much more clean